File: spec/java/src/io/kaitai/struct/spec/TestStrEncodingsEscapingToS.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.StrEncodingsEscapingToS;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
import java.nio.charset.IllegalCharsetNameException;
public class TestStrEncodingsEscapingToS extends CommonSpec {

    @Test
    public void testStrEncodingsEscapingToS() throws Exception {
        final StrEncodingsEscapingToS r = StrEncodingsEscapingToS.fromFile(SRC_DIR + "str_encodings.bin");

        assertUnknownEncoding("ASCII\\\\x", new ThrowingRunnable() {
            @Override
            public void run() throws Throwable {
                r.str1();
            }
        });
        assertUnknownEncoding("UTF-8\\'x", new ThrowingRunnable() {
            @Override
            public void run() throws Throwable {
                r.str2();
            }
        });
        assertUnknownEncoding("SJIS\\\"x", new ThrowingRunnable() {
            @Override
            public void run() throws Throwable {
                r.str3();
            }
        });
        assertUnknownEncoding("IBM437\\nx", new ThrowingRunnable() {
            @Override
            public void run() throws Throwable {
                r.str4();
            }
        });
    }

    private void assertUnknownEncoding(String expectedEncoding, ThrowingRunnable runnable) {
        IllegalCharsetNameException exc = expectThrows(IllegalCharsetNameException.class, runnable);
        assertEquals(exc.getCharsetName(), expectedEncoding);
    }
}

File: spec/python/spec/test_str_encodings_escaping_to_s.py
import unittest
from testformats.str_encodings_escaping_to_s import StrEncodingsEscapingToS

class TestStrEncodingsEscapingToS(unittest.TestCase):
    def test_str_encodings_escaping_to_s(self):
        with StrEncodingsEscapingToS.from_file('src/str_encodings.bin') as r:
            with self.assertRaises(LookupError) as cm:
                r.str1
            self.assertEqual(str(cm.exception), 'unknown encoding: {}'.format("ASCII\\\\x"))

            with self.assertRaises(LookupError) as cm:
                r.str2
            self.assertEqual(str(cm.exception), 'unknown encoding: {}'.format("UTF-8\\'x"))

            with self.assertRaises(LookupError) as cm:
                r.str3
            self.assertEqual(str(cm.exception), 'unknown encoding: {}'.format("SJIS\\\"x"))

            with self.assertRaises(LookupError) as cm:
                r.str4
            self.assertEqual(str(cm.exception), 'unknown encoding: {}'.format("IBM437\\nx"))
